lib/supavisor/db_handler.ex

```diff
@@ -110,6 +110,13 @@ defmodule Supavisor.DbHandler do
         %{tag: :ready_for_query, payload: db_state}, {ps, _} ->
           {ps, db_state}
 
+        %{tag: :backend_key_data, payload: payload}, acc ->
+          key = {data.tenant, self()}
```

Comment:
You may not need to have `self()` as part of the key. Instead you can just use `data.tenant` and use `Registry.values` to look up the values for a given key-pid pair.